src/app/pages/add-new-building-operation/add-new-building-operation.component.ts

```diff
@@ -10,32 +17,129 @@ import { MatSelectModule } from '@angular/material/select'; // Import MatSelectM
+  //     building: ['', Validators.required]
+
+  //   });
 
-  constructor(private formBuilder: FormBuilder) {}
+  //   // this.buildings=(this.myUser as CompanyDTO).PropertyIds;
+  //   //  console.log((this.myUser as CompanyDTO).PropertyIds);
+  //   // // Fetch buildings from the service
+  //   // this.buildingService.getAllBuildings().then(buildings => {
+  //   //   this.buildings = buildings;
+  //   // }).catch(error => {
+  //   //   console.error('Error fetching buildings:', error);
+  //   // });
+  // this.propertiesID=(this.myUser as CompanyDTO).PropertyIds;
 
-  ngOnInit(): void {
+  //  this.propertiesID.forEach(async (id:any)=>{
+  //   this.buildings.push(
+  //    await this.buildingService.getBuilding(id)
+  //   )
+  //  });
+  // }
+
+  async ngOnInit(): Promise<void> {
+    // Initialize the form 
     this.operationForm = this.formBuilder.group({
       operationName: ['', Validators.required],
       description: ['', Validators.required],
       cost: ['', [Validators.required, Validators.pattern(/^\d+(\.\d{1,2})?$/)]],
       building: ['', Validators.required]
     });
+    // Fetch the current Company
+    try {
+      this.myUser = await this.authService.getUser();
+      if (this.myUser) {
+        this.authority = this.myUser.photoURL;
+      } if (this.authority == Authority.Company) {
+        this.userService.getCompanyUser(this.myUser.uid).then((user) => {
+          this.myUser = user;
+        });
+      }
+    } catch (error) {
+      console.error(error);
+      this.authority = '';
+    }
+
+    this.myUser = await this.userService.getCompanyUser(((await this.authService.getUser()) as User)?.uid) as CompanyDTO;
+    this.propertiesID = (this.myUser as CompanyDTO).PropertyIds;
+
+    // Fetch building details for all property IDs asynchronously
+    const fetchBuildingPromises = this.propertiesID.map((id: any) =>
+      this.buildingService.getBuilding(id)
+    );
+
+    // Wait for all building details to be fetched
+    const buildings = await Promise.all(fetchBuildingPromises);
+
+    // Populate the buildings array
+    this.buildings = buildings;
+
+
   }
 
-  // Convenience getter for easy access to form fields
   get f() {
     return this.operationForm.controls;
   }
-
   onSubmit() {
     this.submitted = true;
-
-    // Stop here if form is invalid
+  
     if (this.operationForm.invalid) {
+
       return;
     }
 
-    // Proceed with form submission
-    console.log('Form submitted successfully!');
+    //Create an operation object with form data
+    const operation: Operation = {
+      name: this.operationForm.get('operationName')?.value,
+      description: this.operationForm.get('description')?.value,
+      cost: parseFloat(this.operationForm.get('cost')?.value)
+    };
+
+
+    const selectedBuildingId = this.f['building'].value;
+
+    //Call the addOperation function from BuildingService
+    this.buildingService.addOperation(selectedBuildingId, operation)
```

Comment:
Useless function call since the updateBuilding(building: Building) and the getBuilding(buildingId: string) function already exist